src/games/strategy/triplea/util/TransportUtils.java

```diff
@@ -67,22 +67,37 @@
       final Collection<Unit> transports) {
 
     final List<Unit> canBeTransported = sortByTransportCostDescending(units);
-    final List<Unit> canTransport = sortByTransportCapacityAscendingThenMovesDescending(transports);
+    final List<Unit> canTransport = sortByTransportCapacityDescendingThenMovesDescending(transports);
 
     // Add max units to each transport
     final Map<Unit, Unit> mapping = new HashMap<>();
+    Optional<Unit> finalTransport = Optional.empty();
     for (final Unit transport : canTransport) {
-      int capacity = TransportTracker.getAvailableCapacity(transport);
-      for (final Iterator<Unit> it = canBeTransported.iterator(); it.hasNext();) {
-        final Unit unit = it.next();
-        final int cost = UnitAttachment.get((unit).getType()).getTransportCost();
-        if (capacity >= cost) {
-          capacity -= cost;
-          mapping.put(unit, transport);
-          it.remove();
+
+      // Check if remaining units can all be loaded into 1 transport
+      final int capacity = TransportTracker.getAvailableCapacity(transport);
+      final int remainingCost = getTransportCost(canBeTransported);
+      if (remainingCost <= capacity) {
+        if (!finalTransport.isPresent() || capacity < TransportTracker.getAvailableCapacity(finalTransport.get())) {
+          finalTransport = Optional.of(transport);
         }
+        continue; // skip transports until the smallest one all units fit in with the most moves
+      }
+
+      // Check if we've found the final transport to load remaining units
+      if (finalTransport.isPresent()) {
+        break;
       }
+
+      // Load max units into current transport
+      loadMaxUnits(transport, canBeTransported, mapping);
```

Comment:
consider renaming `transport` to `currentTransport`, and remove the comment that is just restating the code. 
"// Load max units into current transport"